### DISCUSSION TITLE
Where do I put Implemented new features? (example: DelegatePosition class to allow harmonizing coordinate systems)

### DISCUSSION THREAD

**Discussion by DCEM** [microsoft/Qcodes#6694] [Q&A]
Hi,

after I have implemented a feature I would also like to know if i should start a discussion here, start a pull request or create my own module to share the code.
Obviously this is somewhat feature dependend, but I want to know if there is a preffered route.

For example I implemented a few classes to allow for motorized stages to get a harmonizied coordinate sytestem (and collision detection) One class is: DelegatePosition(DelegateParameter) which accepts origin_offset and inverted (amongst others) to translate a position. It will also update its validator to be translated to the corresponding values.
I'm happy to share the code, I'm just not sure whats appropriate way to do it.
If i missed it in the documentation or discussions, I'm sorry.

Kind regards

**Comment by jenshnielsen** [comment]
@DCEM Thanks for your message. Without seeing the code I find it a bit hard to judge it it's a good fit for qcodes or not so I think a pr is a good place to start. I would say that if this is useful for multiple instruments and you are willing to put in the time to ensure that it has good tests, docs and type checking then we would be happy to merge it.

The most important thing to consider when organizing the code is dependencies.
On the instrument side at a highlevel. Qcodes is organized as follows

* qcodes.utilities. / math_utils Something that is generally useful for qcodes but does not it self use qcodes. Cannot import from the rest of qcodes
* qcodes.parameters, Depends on utilities but not anything else
* qcodes.instrument, depends on parameters
* qcodes.extensions. Can depend on anything but is not critically required.

Given your description and that this depends on DelegateParameter this sounds like something that belongs in parameter

**Reply by DCEM** [reply]
@jenshnielsen thank you for your reply. I do unerstand my question was proberbly not specifig enough.

I started with something more basic in #6700 If this is a functionality that is of interest I could do a InterdependetParameter class next.

The issue I encounterd was that Parameters do change dependent on other parameters.

I assume I'm not the only one with this issue and there might be other cases where beeing able to automatically react to a parameter change would be useful.

**Reply by DCEM** [reply]
Ok, i found a better method to implement this: #6731
